lib/network-layer/src/Cardano/Wallet/Network/Implementation/Ouroboros.hs

```diff
@@ -266,29 +266,34 @@ chainSyncFetchNextBlock queue =
 
     -- Ask the node whether it has the block.
     clientStIntersect
-        :: ClientStIntersect block point tip m Void
-    clientStIntersect =
+        :: point -> ClientStIntersect block point tip m Void
+    clientStIntersect point =
         ClientStIntersect
             { recvMsgIntersectFound = \_ _ ->
-                ChainSyncClient $ pure clientRequestBlock
+                ChainSyncClient $ pure $ clientRequestBlock point
             , recvMsgIntersectNotFound = \_ ->
                 result Nothing
             }
 
     -- Request the block in case of success.
     clientRequestBlock
-        :: ClientStIdle block point tip m Void
-    clientRequestBlock =
-        SendMsgRequestNext clientStNext (pure clientStNext)
+        :: point -> ClientStIdle block point tip m Void
+    clientRequestBlock point =
+        SendMsgRequestNext (clientStNext point) (pure $ clientStNext point)
 
     -- Fetch the block on rollforward.
     clientStNext
-        :: ClientStNext block point tip m Void
-    clientStNext = ClientStNext
+        :: point -> ClientStNext block point tip m Void
+    clientStNext point = ClientStNext
         { recvMsgRollForward = \block _ ->
             result $ Just block
-        , recvMsgRollBackward = \_ _ ->
-            result Nothing
+        , recvMsgRollBackward = \point' _ ->
+            -- Note: When starting the chainsync protocol,
+            -- the cardano-node may send an initial rollback message,
+            -- which we will detect here and treat as a no-op.
+            if point' == point
+            then ChainSyncClient $ pure $ clientRequestBlock point
+            else result Nothing
```

Comment:
should that be switched? when `point /= point'` then we sync with requesting the block?

Reply:
No, this is correct: when there is a rollback from `point` to the same `point'`, we ignore the rollback. When the rollback is to someplace else, this means that the `point` is no longer part of the consensus chain, and we need to abort.

Remember, this function is not intended to run the ChainSync protocol properly — instead, it uses ChainSync to fetch one particular block (namely the one after the `point`).

Reply:
aha, thanks! That is correct, indeed!